src/ImageCarouselReact/widget/ImageCarouselReact.tsx

```diff
@@ -0,0 +1,355 @@
+                description: this.descriptionAttr ? itemObj.get(this.descriptionAttr) as string : "",
+                onClick: {
+                    clickMicroflow: this.imageClickMicroflow,
+                    guid: itemObj.getGuid(),
+                    onClickEvent: this.onClickEvent,
+                    page: this.openPage,
+                },
+                url: this.getFileUrl(itemObj.getGuid()),
+            };
+            return data;
+        });
+        callback();
+    }
+    /**
+     * Formats and Returns the object url
+     */
+    private getFileUrl (objectId: string) {
+        logger.debug(this.id + ".getFileUrl");
+        let url: string;
+        if (objectId) {
+            url = "file?target=window&guid=" + objectId + "&csrfToken=" +
+                    mx.session.getCSRFToken() + "&time=" + Date.now();
+        }
+        return url;
+    }
+    /**
+     * iterate over modeler setting of the static images for props and set state
+     */
+    private getDataFromStatic(callback: Function) {
+        logger.debug(this.id  + ".getPropsFromObjects");
+        this.data = this.staticImageCollection.map((itemObj, index) => {
+            let data: Idata;
+            data = {
+                caption: itemObj.imgCaption,
+                description: itemObj.imgDescription,
+                onClick: {
+                    clickMicroflow: itemObj.imageClickMicroflow,
+                    guid: this.contextObj ? this.contextObj.getGuid() : "",
+                    onClickEvent: itemObj.onClickEvent,
+                    page: itemObj.openPage,
+                },
+                url: itemObj.picture,
+            };
+            return data;
+        });
+        callback();
+    }
+
+    // Remove subscriptions
+    private _unsubscribe () {
+        if (this.handles) {
+            for (let handle of this.handles) {
+                mx.data.unsubscribe(handle);
+            }
+            this.handles = [];
+        }
+    }
+    // Reset subscriptions.
+    private _resetSubscriptions () {
+        // only run when widget is using context.
+        if (this.requiresContext) {
+            logger.debug(this.id + "._resetSubscriptions");
+            // Release handles on previous object, if any.
+            this._unsubscribe();
+            // When a mendix object exists create subscribtions.
+            if (this.contextObj) {
+                let objectHandle = mx.data.subscribe({
+                    callback: (guid) => {
+                        logger.debug(this.id + "._resetSubscriptions object subscription update MxId " + guid);
+                        this.updateData(() => {
+                            this.isLoading = false;
+                            this.updateRendering();
+                        });
+                    },
+                    guid: this.contextObj.getGuid(),
+                });
+                this.handles = [objectHandle];
```

Comment:
Design for today! It seems to be a highlander ("there can be only one"). So you can dump the array.